### ISSUE TITLE
Pass waterline errors through as JSON

### ISSUE THREAD

**Issue by kmcgrath** [balderdashy/sails#3051]
When req.wantsJSON is true waterline errors come though as message and stack strings:

    {
      message: <STRING>,
      stack: <STRING>
    }

WLError has the ability to dump itself as:

    { error: 'E_VALIDATION',
      status: 400,
      summary: '2 attributes are invalid',
      model: 'MyModel',
      invalidAttributes: { name: [ [Object], [Object] ], account: [ [Object] ] } }

And it is, up until: https://github.com/balderdashy/sails/blob/a210b48667708bea687dd5ec61ce9e07ffc0c005/lib/hooks/responses/index.js#L221

where a check for a `message` and `stack` key cause that nice JSON object to be replaced by string version.

Can an extra check be added to see if the typeof is WLError and pass the JSON through?  Would that break something I'm not thinking about?

**Comment by leobudima** [comment]
+1, I'm trying to get an error message from model lifecycle callbacks up to the client in a way that can be handled and displayed nicely to the user...

**Comment by sailsbot** [comment]
Thanks for posting, @kmcgrath.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
